app/controllers/exercises_controller.rb

```diff
@@ -33,6 +33,7 @@ def start!
 
   def set_default_content!
     @files = @exercise.files_for(current_user)
+    @highlight_modes = Language.all.map { |it| { extension: it.extension, highlight_mode: it.highlight_mode } }
```

Comment:
Please avoid adding extra state to the controllers and views, since it is difficult to trace in ruby - since the dynamic nature of variables - where it was declared and when it is mutated.  

I'd recommend declaring it as a helper method either in this class or in a `MultipleFileEditorHelper`. I'd prefer the latter